src/sas/qtgui/Utilities/TabbedModelEditor.py

```diff
@@ -426,6 +464,38 @@ def updateFromEditor(self):
         msg = filename + " successfully saved."
         self.parent.communicate.statusBarUpdateSignal.emit(msg)
         logging.info(msg)
+        self.regenerateDocumentation()
+    
+    def regenerateDocumentation(self):
+        """
+        Defer to subprocess the documentation regeneration process
+        """
+        sas_path = os.path.abspath(os.path.dirname(sys.argv[0]))
+        recompile_path = GuiUtils.RECOMPILE_DOC_LOCATION
+        regen_docs_location = sas_path + "/" + recompile_path + "/makedocumentation.py"
+        d = threads.deferToThread(self.regenerateDocsCommand, regen_docs_location) # Regenerate specific documentation file
+        d.addCallback(self.emitDocsRegenerated)
+
+    def regenerateDocsCommand(self, regen_docs_location):
+        """
+        Regenerates documentation for a specific file (target) in a subprocess
+        """
+        import subprocess
+        # Pass in file in correct format if the file is a model or a native rst
+        if "models" in self.file_to_regenerate:
+            pass_in_file = os.path.basename(self.file_to_regenerate)
+        else:
+            pass_in_file = self.file_to_regenerate
+        command = [
+            sys.executable,
+            regen_docs_location,
+            pass_in_file,
+        ]
+        doc_regen_dir = os.path.dirname(regen_docs_location)
+        subprocess.run(command, cwd=doc_regen_dir) # cwd parameter tells subprocess to open from a specific directory
```

Comment:
I didn't look in detail at the code but it seems familiar. Can you collect all your interactions with sphinx in one module?